// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: list_entities.proto

package identityd2

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// define the regex for a UUID once up-front
var _list_entities_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on ListEntitiesRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListEntitiesRequest) Validate() error {
	if m == nil {
		return nil
	}

	if v, ok := interface{}(m.GetEntity()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListEntitiesRequestValidationError{
				field:  "Entity",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// ListEntitiesRequestValidationError is the validation error returned by
// ListEntitiesRequest.Validate if the designated constraints aren't met.
type ListEntitiesRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListEntitiesRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListEntitiesRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListEntitiesRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListEntitiesRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListEntitiesRequestValidationError) ErrorName() string {
	return "ListEntitiesRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListEntitiesRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListEntitiesRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListEntitiesRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListEntitiesRequestValidationError{}

// Validate checks the field values on ListEntitiesResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListEntitiesResponse) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetEntities() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListEntitiesResponseValidationError{
					field:  fmt.Sprintf("Entities[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// ListEntitiesResponseValidationError is the validation error returned by
// ListEntitiesResponse.Validate if the designated constraints aren't met.
type ListEntitiesResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListEntitiesResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListEntitiesResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListEntitiesResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListEntitiesResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListEntitiesResponseValidationError) ErrorName() string {
	return "ListEntitiesResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListEntitiesResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListEntitiesResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListEntitiesResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListEntitiesResponseValidationError{}
